File: backend/Blogigg.Domain/Entities/Post.cs
namespace Bloggig.Domain.Entities;

public class Post
{
    public Guid Id { get; set; }
    public string Title { get; set; }
    public string Content { get; set; }
    public Guid AuthorId { get; set; }
    public string ThumbnailUrl { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public string Status { get; set; }
    public List<Tag> Tags { get; set; } = new List<Tag>();
}

File: frontend/src/app/interfaces/Post.ts
import Tag from "./Tag";
import User from "./User";

export default interface Post
{
  id: string,
  title: string,
  content: string,
  authorId: string,
  author: User,
  thumbnailUrl: string,
  createdAt: Date,
  updatedAt: Date,
  status: string,
  tags: Tag[]
} 